scipy/optimize/optimize.py

```diff
@@ -693,7 +693,15 @@ def _approx_fprime_helper(xk, f, epsilon, args=(), f0=None):
     for k in range(len(xk)):
         ei[k] = 1.0
         d = epsilon * ei
-        grad[k] = (f(*((xk + d,) + args)) - f0) / d[k]
+        df = (f(*((xk + d,) + args)) - f0) / d[k]
+        if not np.isscalar(df):
+            try:
+                df = df.item()
```

Comment:
This could also raise `AttributeError`. 

Also, I'm not quite sure we want this check. Returning a scalar is not that hard, why complicate things why allowing 0-D arrays?

Reply:
Right, it should also catch that `AttributeError`. I'll add that.

As for the 0-D array check: many of the tests in `test_optimize.py` set the initial condition `x0` to be a 0-D array (for example, [here](https://github.com/scipy/scipy/blob/master/scipy/optimize/tests/test_optimize.py#L175)). So, without this check those tests failed.

Reply:
ah yes, makes sense - okay let's keep it